File: Code Generation/responses/cpp/gemini/easy/transform-array-by-parity.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> transformArray(vector<int> nums) {
        for (int i = 0; i < nums.size(); ++i) {
            // Replace even numbers with 0, odd numbers with 1
            if (nums[i] % 2 == 0) {
                nums[i] = 0;
            } else {
                nums[i] = 1;
            }
        }
        
        // Sort the modified array in non-decreasing order
        sort(nums.begin(), nums.end());
        
        return nums;
    }
};

File: Code Generation/responses/python/deepseek/easy/transform-array-by-parity.py
from typing import List

class Solution:
    def transformArray(self, nums: List[int]) -> List[int]:
        # Replace each even number with 0 and each odd number with 1
        transformed = [0 if num % 2 == 0 else 1 for num in nums]
        
        # Sort in non-decreasing order
        transformed.sort()
        
        return transformed